eclipse/set_memory_flags.py

```diff
@@ -0,0 +1,82 @@
+# SPDX-FileCopyrightText: Copyright DB InfraGO AG and contributors
+# SPDX-License-Identifier: Apache-2.0
+
+import os
+import pathlib
+
+
+def remove_all_memory_flags(file_content: list[str]) -> list[str]:
+    """Remove all memory flags from *.ini files.
+
+    To make sure that we don't have any conflicting memory flags in the
+    .ini file, we remove all flags that start with -Xm.
+
+    This will explicitely remove the -Xms and -Xmx flags.
+    """
+
+    return [
+        line for line in file_content if not line.strip().startswith("-Xm")
+    ]
+
+
+def set_memory_flags(
+    file_content: list[str], memory_min: str, memory_max: str
+) -> None:
+    """Set the memory flags in *.ini.
+
+    If the value of memory_max ends with a %, we assume that it's a percentage value.
+    - If used in a Kubernetes cluster, it determines the values from the Pod requests/limits.
+    - If used on a host system, it determines the value from the host memory.
+
+    See Also
+    --------
+    - https://stackoverflow.com/a/65327769
+    - https://www.merikan.com/2019/04/jvm-in-a-container/#java-10
+    """
+
+    if memory_min.strip().endswith("%"):
+        append_flag_to_file(
+            file_content, "XX:InitialRAMPercentage=", memory_min.strip("%")
+        )
+        append_flag_to_file(
+            file_content, "XX:MinRAMPercentage=", memory_min.strip("%")
+        )
+    else:
+        append_flag_to_file(file_content, "Xms", memory_min)
+
+    if memory_max.strip().endswith("%"):
+        append_flag_to_file(
+            file_content, "XX:MaxRAMPercentage=", memory_max.strip("%")
+        )
+    else:
+        append_flag_to_file(file_content, "Xmx", memory_max)
+
+
+def print_vm_memory_usage_during_start(file_content: list[str]) -> None:
+    """Print the JVM memory options during start."""
+
+    file_content.append("-XshowSettings:vm")
```

Comment:
A dedicated function id a bit an overkill here (see comment on line 80)

Reply:
I'd not call it "overkill", it's a practice proposed by "Clean code" and I personally find code a lot more readable when code is outsources to functions, which describe what they do. I'm usually not interested in how it's done, I just want to know what is done unless I'm interesting in the function - and in this case I can easily jump to it with IDEs.

The goal is to reduce verbosity from the main function to keep it readable and hide the details in the function.

Reply:
Interesting view. With that argument I'm not sure what I prefer. Fine that it'll be left as is here.